Jenkinsfile

```diff
@@ -1,128 +1,145 @@
+        throw e
+      }
+
+      try
+      {
+        stage('Test') {
+          withEnv(mvnEnv) {
+            timeout(time: 90, unit: 'MINUTES') {
+              // Run test phase / ignore test failures
+              sh "mvn -V -B install -Dmaven.test.failure.ignore=true -Prun-its"
+              // Report failures in the jenkins UI
+              step([$class: 'JUnitResultArchiver',
+                    testResults: '**/target/surefire-reports/TEST-*.xml'])
+              // Collect up the jacoco execution results
+              def jacocoExcludes =
+                      // build tools
+                      "**/org/eclipse/jetty/ant/**" +
+                              ",**/org/eclipse/jetty/maven/**" +
+                              ",**/org/eclipse/jetty/jspc/**" +
+                              // example code / documentation
+                              ",**/org/eclipse/jetty/embedded/**" +
+                              ",**/org/eclipse/jetty/asyncrest/**" +
+                              ",**/org/eclipse/jetty/demo/**" +
+                              // special environments / late integrations
+                              ",**/org/eclipse/jetty/gcloud/**" +
+                              ",**/org/eclipse/jetty/infinispan/**" +
+                              ",**/org/eclipse/jetty/osgi/**" +
+                              ",**/org/eclipse/jetty/spring/**" +
+                              ",**/org/eclipse/jetty/http/spi/**" +
+                              // test classes
+                              ",**/org/eclipse/jetty/tests/**" +
+                              ",**/org/eclipse/jetty/test/**";
+              step([$class: 'JacocoPublisher',
+                    inclusionPattern: '**/org/eclipse/jetty/**/*.class',
+                    exclusionPattern: jacocoExcludes,
+                    execPattern: '**/target/jacoco.exec',
+                    classPattern: '**/target/classes',
+                    sourcePattern: '**/src/main/java'])
+              // Report on Maven and Javadoc warnings
+              step([$class: 'WarningsPublisher',
+                    consoleParsers: [
+                            [parserName: 'Maven'],
+                            [parserName: 'JavaDoc'],
+                            [parserName: 'JavaC']
+                    ]])
+            }
+            if(isUnstable())
+            {
+              notifyBuild("Unstable / Test Errors", jdk)
+            }
+          }
         }
+      } catch(Exception e) {
+        notifyBuild("Test Failure", jdk)
+        throw e
       }
-    }
-  } catch(Exception e) {
-    notifyBuild("Test Failure")
-    throw e
-  }
 
-  try
-  {
-    stage 'Compact3'
+      try
+      {
+        stage ('Compact3') {
 
-    dir("aggregates/jetty-all-compact3") {
-      withEnv(mvnEnv) {
-        sh "mvn -B -Pcompact3 clean install"
+          dir("aggregates/jetty-all-compact3") {
+            withEnv(mvnEnv) {
+              sh "mvn -V -B -Pcompact3 clean install"
```

Comment:
Does `compact3` style releases still make sense in Java9 or Java10 ?

Reply:
for we can ignore that (maybe adding a profile activated with java8?)